core-bundle/src/DependencyInjection/ContaoCoreExtension.php

```diff
@@ -48,6 +48,10 @@ public function getConfiguration(array $config, ContainerBuilder $container): Co
 
     public function load(array $configs, ContainerBuilder $container): void
     {
+        if ('UTF-8' !== $container->getParameter('kernel.charset')) {
+            trigger_error(sprintf('Using the charset "%s" is not supported by Contao, use "UTF-8" instead.', $container->getParameter('kernel.charset')), E_USER_WARNING);
```

Comment:
Shouldn't we throw an exception here? What am I supposed to do with a warning if nothing works anyway if I don't change it? :D

Reply:
Shouldn't we check this only when booting the Contao framework?

Reply:
I’m not sure. I think we should probably trigger a deprecation instead and throw an exception in Contao 5.

Reply:
>  if nothing works anyway if I don't change it?

Is this really the case? Because that would be a BC break.

Reply:
@ausi tried it in https://github.com/contao/contao/issues/3050#issuecomment-854714950 - Contao must have never supported anything other than utf-8 for years.

Reply:
Then I guess it does not matter whether we throw an exception or trigger a deprecation warning. 😄

Reply:
Done in 8b209874030e309e9d3eb70be60c9f3858c6ad85.